### ISSUE TITLE
Trying to transform a sketch activates sketch edit mode instead of transform dialog

### ISSUE THREAD

**Issue by maxwxyz** [FreeCAD/FreeCAD#18296]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Trying to transform a Body with a Sketch by selecting the sketch in the 3D view and right click > Transform does not start the transform tool command but enters sketch edit more.

Steps to reproduce:
1. Create a body
2. Create a sketch with geometry
3. Leave sketch edit mode
4. Select the sketch in the 3D view; right click and choose `Transform`
5. Sketch edit mode is launched

Could be related to the various issues with the edit modes.

### Full version info

```shell
OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.1.0dev.39416 (Git) Conda
Build type: Release
Branch: main
Hash: f6268ffd2877dadb4440ade541e12c388a517d6e
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: German/Germany (de_DE)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * CfdOF 1.29.1
  * Curves 0.6.51
  * dodo 1.0.1
  * fasteners 0.5.31
  * freecad.gears 1.3.0
  * Manipulator 1.5.7
  * OpenTheme 2024.9.1
  * OpticsWorkbench 1.0.26
  * Rocket 4.0.1
  * sheetmetal 0.5.6
  * Silk 0.1.5
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
@FEA-eng also when trying to transform FEM results pipeline, the task view is opened to edit the results pipeline (not that this would make sense to transform this)

**Comment by FEA-eng** [comment]
Still relevant

**Comment by VictorLDunn** [comment]
Discovered this bug using FC 1.0.2 AppImage for myself a few days ago whilst trying to rotate a sketch a few degrees about the z axis.

Because the sketch was a simple rectangle for a part>slice - making a face from the sketch, then allowed the face to be transformed to the correct orientation. 

Bug also existed in FC 0.21.2,

And still exists in the latest Version: 
1.1.0dev.20251031 (Git shallow) AppImage
Build date: 2025/10/31 03:23:50

If a sketch has no attachment - then being able to Transform would be useful (compared to having to trying to use placement properties !)

If the sketch is attached then selecting Transform could either flag a warning, or open the attachment editor  

So, please keep this issue active for a fix sometime in the future.

**Comment by kaiwas** [comment]
sadly

**Comment by maxwxyz** [comment]
@PaddleStroke there is a fix for this Sketcher issue at https://codeberg.org/xCAD/FreeCAD11/pulls/142/ if you want to cherry-pick.

**Comment by PaddleStroke** [comment]
I'll check that now

**Comment by PaddleStroke** [comment]
Why sketch do not have the "transform" option in the tree right click?

**Comment by maxwxyz** [comment]
Not sure, usually in PD bodies they need an attachment so the transform command in Part Design should be rerouted to the body (same as with every PD feature). With unattached sketches e.g. in Part it should be able to transform them.
You can select the sketch and activate the transform tool by right clicking in the 3D view

**Comment by PaddleStroke** [comment]
Werner's gone and forked the project then. And they reverted the glorious formating we merged recently. So this is bound to diverge very quickly now.

Such a shame to see that all of work go to waste. Someone's in charge of cherry picking Werner's patches?

**Comment by PaddleStroke** [comment]
@kadet1090  FYI with the total formating, cherry-picking patches from Werner is going to be a huge PITA.

I had to port the fix manually.

**Comment by maxwxyz** [comment]
@3x380V FYI

**Comment by PaddleStroke** [comment]
@maxwxyz I already ported this fix in [#27063](https://github.com/FreeCAD/FreeCAD/pull/27063)

**Comment by kadet1090** [comment]
> [@kadet1090](https://github.com/kadet1090) FYI with the total formating, cherry-picking patches from Werner is going to be a huge PITA.
> 
> I had to port the fix manually.

@PaddleStroke It indeed requires more steps, but you can automate this process pretty well - here is a handy script with every command explained:

```
#!/usr/bin/env bash

COMMIT_TO_CHERRY_PICK="$1"

# First reset to a commit BEFORE one that we need to cherry pick
git reset --hard "$COMMIT_TO_CHERRY_PICK~"

# Cherry pick the formatting update
git cherry-pick b70b312

# Apply the formatting - it requires multiple runs because clang-format is not ideal
while true; do
    pre-commit run --all-files && break
done

# Create scrap commit that contains formatting changes
git commit -am "chore: Apply formatting changes"

# Cherry pick the actual changes - they will probably conflict, but the conflicts will be only on formatting
git cherry-pick "$COMMIT_TO_CHERRY_PICK"

# Use files from the cherry-picked commit (i.e. --theirs) as-is, without manual conflict solving
git diff --name-only --diff-filter=U --relative | xargs git checkout --theirs -- 

# Apply the formatting - one time should be sufficient, but run in loop to be safe
while true; do
    pre-commit run --all-files && break
done

# Finish the cherry-picking
git add .
git cherry-pick --continue
```

This will give you as output something like this:

```
[main e2ddc58ad8] Sketch: Forward non-default edit mode to base class
 Author: wwmayer <wmayer@freecad.org>
 Date: Tue Jan 13 14:02:44 2026 +0100
 6 files changed, 50 insertions(+), 31 deletions(-)
```

The e2ddc58ad8 commit (I've pushed it to my fork)  will contain the same changeset but will be based on reformatted codebase - you can then cherry-pick it and solve actual code conflicts with it without having to deal with formatting changes. 

@3x380V FYI, but you probably have a workflow for it already.

**Comment by VictorLDunn** [comment]
Working with Part WB and off axis sketches within many of my projects, this is good news. Thank you everyone.

**Comment by 3x380V** [comment]
@kadet1090, @PaddleStroke, Werner's original https://github.com/wwmayer/FreeCAD/tree/main-dev branch is rebased at https://github.com/3x380V/FreeCAD/tree/wmayer where you can see the current status as well... Before each original hash is a mark, where `[x]` means patch was merged to FreeCAD and PR is mentioned in parentheses; `[-]` means patch was dropped with alternative PR in parentheses. `[ ]` means patch is pending. So nothing was really wasted, but awaits someone to pick the patches and open PR. The branch is up to date and compiles. Btw, no matter how hard I had to massage the patches while rebasing, Werner's authorship was maintained unlike commit merged to fix this bug.

As for the recent @wwmayer's patches at codeberg, I do not maintain any updated branch of them as I'm using the fork directly...

**Comment by PaddleStroke** [comment]
> Werner's authorship was maintained unlike commit merged to fix this bug.

Why do yo usay that? The commit has werner's authorship

<img width="578" height="181" alt="Image" src="https://github.com/user-attachments/assets/3c1cccc3-0cf5-44da-bb98-74ce13a50a4b" />

**Comment by PaddleStroke** [comment]
> So nothing was really wasted, but awaits someone to pick the patches and open PR. The branch is up to date and compiles.

Perhaps you could do a single PR with all the patches. It's not like Werner's coding skills are doubtfull we can just merge at once

**Comment by 3x380V** [comment]
> Why do yo usay that? The commit has werner's authorship

Well, the `main` branch disagrees: https://github.com/FreeCAD/FreeCAD/commit/93abfc4fa4e8b1c055cdcae23dd9e76f15f815ef.patch
```
From: PaddleStroke <pierrelouis.boyer@gmail.com>
Date: Wed, 21 Jan 2026 22:41:44 +0100
Subject: [PATCH] Sketch: Forward non-default edit mode to base class. (#27063)
```

> Perhaps you could do a single PR with all the patches. It's not like Werner's coding skills are doubtfull we can just merge at once

There are also patches where Werner disagrees with the rest of developers, so it needs some sorting first anyway. I was hoping developers working on specific areas would cherry-pick the readily available patches and attempt to merge them...

**Comment by PaddleStroke** [comment]
> Well, the main branch disagrees

Hmm well I don't know what happened, I'm not so fluent in git. On the PR the commit showed clearly Werner's authorship.

Here is the command I used to commit, is this not how it's supposed to be done?

`git commit --author="wwmayer <wmayer@freecad.org>" -m "Sketch: Forward non-default edit mode to base class" -m "Ported from Codeberg commit 66dbf277"`

**Comment by 3x380V** [comment]
@PaddleStroke, indeed your original commit https://github.com/FreeCAD/FreeCAD/pull/27063/commits/09a5b70650bea87bf0f5d443b24cb4705ecd6a48 is correct. I do not know what happened then either, I was just merely pointing at the result.

**Comment by kadet1090** [comment]
@3x380V The authorship might be my (githubs?) misdoing - I may have used the "Squash" to merge as github remembers last used option and it might have done something with authorship. I'll try by more careful with not using squash for such commits as I have not assumed it may have an effect on authorship.

**Comment by 3x380V** [comment]
@kadet1090, yes that seems to be a cause. It happened in the past and I had a discussion with @chennes (probably as I cannot find it anymore) to avoid squash merges completely. If I remember the outcome correctly, idea was refused as it rises demands on developers - they would be expected to rebase and eventually squash commits on their own. Still, I think it is reasonable to expect and require well prepared PRs ;-)
